### PR TITLE
feat: migrate resource group tagging interceptor to V2

### PR DESCRIPTION
**Pull request by ChenTanyi**

### CODE REVIEW

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                byteBuffer -> {
+                    byte[] body = new byte[byteBuffer.remaining()];
+                    byteBuffer.get(body);
+                    String bodyStr = new String(body, StandardCharsets.UTF_8);
+
+                    ResourceGroupInner resourceGroupInner;
+                    try {
+                        resourceGroupInner = adapter.deserialize(bodyStr, ResourceGroupInner.class, SerializerEncoding.JSON);
+                        if (resourceGroupInner == null) {
+                            throw new IOException();
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java]: Do not use exception as flow control.

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+
+
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java]: Clean up a bit.

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                        return Mono.error(new RuntimeException("Failed to deserialize " + bodyStr));
+                    }
+
+                    Map<String, String> tags = resourceGroupInner.getTags();
+                    if (tags == null) {
+                        tags = new HashMap<>();
+                    }
+                    tags.put("product", "javasdk");
+                    tags.put("cause", "automation");
+                    tags.put("date", Instant.now().toString());
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java]: Will it overwrite users with same name?

**Comment by ChenTanyi** [comment]
@weidongxu-microsoft The V1 interceptor is just the deletion file.
Or you can see it at https://github.com/azure/azure-libraries-for-java/blob/7ef6e15387864cf3b41961583b580d8f570ce9aa/azure-mgmt-resources/src/test/java/com/microsoft/azure/management/resources/core/ResourceGroupTaggingInterceptor.java

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java (hunk 3 above)
Comment (ChenTanyi) [reply azure-mgmt-resources/src/test/java/com/azure/management/resources/core/ResourceGroupTaggingPolicy.java]: Yes, so it is only used at unit test.

**Comment by ChenTanyi** [comment]
@weidongxu-microsoft It adds tags only for unit test, so may not be suitable to put in the ResourceGroupImpl.

**Comment by weidongxu-microsoft** [comment]
Then you are free to go if it is only for testing.

Please still fix the exception part. One should not use exception for normal if/else handling.

**Comment by ChenTanyi** [comment]
> Then you are free to go if it is only for testing.
> 
> Please still fix the exception part. One should not use exception for normal if/else handling.

Done

**Pull request merged**